Update Houdini addon docs (#83)
ynput/ayon-documentation: website/docs/admin_hosts_houdini.md

Enabling this setting will cause Houdini creator to add a **self publish button** when creating new publish instances. More Info on how how the **self publish button** works [here](artist_hosts_houdini#ayon-publishing-process)

![general_settings_self_publish_button](assets/houdini/admin/general_settings_self_publish_button.png)

### Update Houdini Vars on context change

Allows admins to have a list of vars (e.g. JOB) with (dynamic) values that will be updated on context changes, e.g. when switching to another asset or task.

:::info
**JOB** Var: If `JOB` is present in the Houdini var list and has an empty value, Ayon will set its value to `$HIP`
:::

![update-houdini-vars-context-change](assets/houdini/admin/update-houdini-vars-context-change.png)

1. **Update Houdini vars on context change :** Disabling this setting will leave all Houdini vars unmanaged and thus no context update changes will occur.
2. **(+) :** add new Houdini variable
3. **Var :** The Houdini variable you want to add. No need to include `$`.
  > For consistency reasons we always force all vars to be uppercase. e.g. `myvar` will be `MYVAR`
4. **Value :** The value you want Ayon to maintain. 

>   > Using template keys is supported but formatting keys capitalization variants is not, e.g. `{Asset}` and `{ASSET}` won't work. For a list of available template keys go to [available template keys](#what-template-keys-are-available-for-use-in-houdini-settings)

Reviewer: ```suggestion
  > Using template keys is supported but formatting keys capitalization variants is not, e.g. `{Asset}` and `{ASSET}` won't work. For a list of available template keys go to [available template keys](#what-template-keys-are-available-for-use-in-houdini-settings).
```

  > If the folder does not exist on the context change it will be created by this feature so that the path will always try to point to an existing folder.

## Color Management (ImageIO)

Allows admins to override the global color management settings, check [Host specific overrides](https://ayon.ynput.io/docs/admin_colorspace#host-specific-overrides)

![houdini-color-management](assets/houdini/admin/houdini-color-management.png)

## Shelves Manager

Allows admins to add studio's [Shelves](https://www.sidefx.com/docs/houdini/shelf/index.html) and scripts into Houdini. 

### Different Modes

Shelves manager works in two modes: 
1. Add a `.shelf` file. you are required only to fill in the `Shelf Set Path`.
  ![houdini_shelves_manager_mode_1](assets/houdini/admin/houdini_shelves_manager_mode_1.png)

2. Add Scripts and Ayon will create a shelf for you.
  > *Currently, Ayon just adds the new shelf definition in `default.shelf`*
  
  ![houdini_shelves_manager_mode_2](assets/houdini/admin/houdini_shelves_manager_mode_2.png)

The Shelf Set Path is used to load a .shelf file to generate your shelf set. If the path is specified, you don't have to set the shelves and tools.

> ### Shelves Example

Reviewer: The shelves can get a bit better description that it's **either** the shelf set path or the scripts below for a particular shelf set.

So we could word it e.g. like this:

> You have two options to define a shelf which are mutually exclusive from one another:
> ﻿
> - If you set a Shelf Set Path that shelf is loaded, any scripts defined below it are ignored.
> - If you have **no** Shelf Set Path, then it'll create a shelf dynamically from the Shelf Scripts.

:::info

> Using template keys is supported but formatting keys capitalization variants is not, e.g. `{Asset}` and `{ASSET}` won't work. For a list of available template keys go to [available template keys](#what-template-keys-are-available-for-use-in-houdini-settings)

Reviewer: ```suggestion
Using template keys is supported but formatting keys capitalization variants is not, e.g. `{Asset}` and `{ASSET}` won't work. For a list of available template keys go to [available template keys](#what-template-keys-are-available-for-use-in-houdini-settings).
```

:::

In this example I made two shelves: 

> 1. `Null Shelf` which defined in my `null_shelf.shelf` file

Reviewer: ```suggestion
1. `Null Shelf` which gets loaded from my pre-defined `null_shelf.shelf` file.
```

> 2. `Ayon Shelf` which includes my script `Say Hello`

Reviewer: ```suggestion
2. `Ayon Shelf` which includes my script `Say Hello`.
```

sidebar_label: Houdini
---

## General
### Update Houdini Vars on context change

Allows admins to have a list of vars (e.g. JOB) with (dynamic) values that will be updated on context changes, e.g. when switching to another asset or task.

Using template keys is supported but formatting keys capitalization variants is not, e.g. `{Asset}` and `{ASSET}` won't work

:::note

> If `Treat as directory` toggle is activated, Ayon will consider the given value is a path of a folder.

Reviewer: ```suggestion
If `Treat as directory` toggle is activated, Ayon will consider the given value as a path of a folder.
```

_consider as_ or _assume is_

---

## FAQ

### When will we need to modify Houdini addon code ?
We've spent a lot of time making it possible to customize Houdini addon from settings.
e.g. you can add your studio's menus, tools and digital assets from settings. 

Still, you may need to modify Ayon addons only when :  
- Adding/Modifying/Customizing Launch Hooks
- Registering new Callbacks
- Adding/Modifying/Customizing plugins (create, publish, load, Actions)
- Adding a new Ayon launcher tray tool

### How to add my menus, tools and digital assets to Houdini ?
Mostly, you'd only need to update Environment to share these tools with your team. 
e.g., 
- `PYTHONPATH` to add a custom python library path
- `HOUDINI_MENU_PATH` to add a custom menu path
- `HOUDINI_OTL_PATH` to add a custom digital assets path

Ayon allows you to add environment variables as:
1. Global environment variable (it wil be available for all DCCs)
2. Application specific (it wil be available for a particular DCC)
3. Application variant specific (it wil be available for a particular DCC version)
4. Tool (it wil be available on demand per project per asset from project Editor)

> For a detailed guide visit: [Ayon/Openpype Env Vars and Tools Configuration Explained](https://community.ynput.io/t/openpype-env-vars-and-tools-configuration-explained/540)

Reviewer: Do we recommend linking to community forum posts or is the end goal to have the crucial information from topics like those to be covered in the docs itself?

Author: The end goal is not discussed yet.
however, I did so as I believe FAQ should have simple quick answers and the full details should live some where else. 
I used the community link because it's updated and already contains the full answer.

Reviewer: I'd say it's a perfect step for now if there are no better resources - but Community topics can grow very large still making it hard to find the right info over time.

11. **⇅:** change the **shelf** order into the **shelf set**.
12.  **⇅:** change the **tool** order in it **shelf**.

## Creator plugins
Enable or disable the plugins. Some of them have extra options such as defining the default subsets names.

> **Plugins list:** Create Alembic Camera, Create Arnold Ass, Create Arnold ROP, Create Composite (Image Sequence), Create Houdini Digital Asset, Create Karma ROP, Create Mantra ROP, Create PointCache (Abc), Create PointCache (Bgeo), Create Redshift Proxy, Create Redshift ROP, Create Review, Create Static Mesh, Create USD (experimental), Create USD render (experimental), Create VDB Cache, Create VRay ROP.

Reviewer: ```suggestion
**Plugins list:**

- Create Alembic Camera
- Create Arnold Ass
- Create Arnold ROP
- Create Composite (Image Sequence)
- Create Houdini Digital Asset
- Create Karma ROP
- Create Mantra ROP
- Create PointCache (Abc)
- Create PointCache (Bgeo)
- Create Redshift Proxy
- Create Redshift ROP
- Create Review
- Create Static Mesh
- Create USD (experimental)
- Create USD render (experimental)
- Create VDB Cache
- Create VRay ROP

```

> ## Publish plugings

Reviewer: ```suggestion
## Publish plugins
```

Enable or disable the plugins executed at publishing.

**Publish plugins list:** Collect Rop Frame Range, Validate Latest Containers, Validate Mesh is Static, Validate Review Colorspace, Validate Subset Name, Validate Unreal Static Mesh Name,Validate workfile paths settings.

### Collect Rop Frame Range

Disable this if you want the publisher to ignore start and end handles specified in the asset data for publish instances.
> Artists are allowed to override this value in the publisher UI.

![houdini_publish_plugins_collectors_frame_range](assets/houdini/admin/houdini_publish_plugins_collectors_frame_range.png)

---

## FAQ

### When will I need to modify Houdini addon code ?
We've spent a lot of time making it possible to customize Houdini addon from settings.
e.g. you can add your studio's menus, tools and digital assets from settings. 

Still, you may need to modify Ayon addons only when :  
- Adding/Modifying/Customizing Launch Hooks
- Registering new Callbacks
- Adding/Modifying/Customizing plugins (create, publish, load, Actions)
- Adding a new Ayon launcher tray tool

### How to add my menus, tools and digital assets to Houdini ?

> Consider adding new shelves firstly as you can use them to share a lot of tools with your team.

Reviewer: What's the "firstly" intend to say? :)

```suggestion
First, consider adding custom shelves as you can use easily use them to share a lot of tools with your team.
```

However, if you need to add custom python libraries or custom OTLs or custom menus, updating environment variables is your way to make them available to your team. 

e.g., 
- `PYTHONPATH` to add a custom python library path
- `HOUDINI_MENU_PATH` to add a custom menu path
- `HOUDINI_OTL_PATH` to add a custom digital assets path

Ayon allows you to add environment variables as:
1. Global environment variable (it wil be available for all DCCs)
2. Application specific (it wil be available for a particular DCC)
3. Application variant specific (it wil be available for a particular DCC version)

> 4. Tool (it wil be available on demand per DCC per project per asset)

Reviewer: ```suggestion
4. Tool (it will be available on demand per DCC per project per asset)
```

For a detailed guide visit: [Ayon/Openpype Env Vars and Tools Configuration Explained](https://community.ynput.io/t/openpype-env-vars-and-tools-configuration-explained/540)

### How to publish lookdev from Houdini ? 

Publishing and managing Lookdev in Houdini is within our plan.

> The current solution to publish materials is to use HDAs as you can publish most of Houdini nodes as hda.

Reviewer: ```suggestion
The current solution to publish materials is to use HDAs as you can publish most of Houdini nodes as HDA.
```
